### PR TITLE
Pipelining followups

### PR DESCRIPTION
**Pull request by tzaffi**
# Summary

Addressing the [Pipelining Followups Issue](https://github.com/algorand/conduit/issues/141) along with the "CLI health endpoint bug" which I discovered while working on the block generator.

## The CLI Health Endpoint Bug
This bug was actually introduced by the [Pipelining PR](https://github.com/algorand/conduit/pull/128) which preceded the health endpoint, and in particular [this line](https://github.com/tzaffi/conduit/blob/9918073c60effa45f05c9266e53f500a5d672410/conduit/pipeline/pipeline.go#L684) which caused `Start()` itself to wait forever. As a consequence, the setup of the health endpoint was never reached while conduit was running.

## Issue

#141 

## TODO's

- [x] remove `retriesNoOutput()` and its test, unless a use case for _this PR_ comes up.

## Testing

Introducing `TestHealthEndpoint()` in `cli_test.go` for checking that the health endpoint is ON/OFF as configured. I verified that when the bug is re-introduced, the `API_ON` case fails as expected. For the purposes of the test, I also introduced a `noop importer` which waits 100 ms before returning in its `GetBlock()` method.

### CODE REVIEW

File: conduit/pipeline/common.go
Comment (tzaffi) [reply conduit/pipeline/common.go:80]: @algochoi - I had originally added this because I wanted to retry the call to `GetGenesis()` (which takes no params) inside of the pipeline's `Init()`. But then I realized that retrying at that point is too late and you'd really have have the importer do that (and it can't access this package).

UPSHOT: I commit to deleting this function and its test before merging, unless there's a use case that crops up mighty soon.

File: conduit/pipeline/pipeline.go
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:700]: This is the health endpoint bugfix.

File: conduit/plugins/importers/noop/noop_importer.go
Comment (winder) [inline conduit/plugins/importers/noop/noop_importer.go:64]: Why the sleep?

**Review by winder — APPROVED** [review]
Approved pending your offer to remove the unused function.

File: conduit/plugins/importers/noop/noop_importer.go
Comment (tzaffi) [reply conduit/plugins/importers/noop/noop_importer.go:64]: Just to slow down the pipeline. For example, if you have a no-op for your importer and a file-writer for your exporter this limits the output to 10 blocks per second which is a little easier to manage than without having any sort of brakes on. Another option is to make this configurable, but that didn't strike me as ideal either since I'd like the no-op importer to have no config. LMK.

**Comment by tzaffi** [comment]
> Approved pending your offer to remove the unused function.

d6358415c5c2cd598662697ab829b44977e7c3b3

**Review by algochoi — APPROVED** [review]

**Review by winder — APPROVED** [review]

**Pull request merged**
